[13.0][FIX] account_financial_report: Don't include unaffected account if accounts are filtered

joao-p-marques opened the pull request:
When generating a General Ledger or a Trial Balance, if some accounts are already filtered in the wizard, don't show the information related to the unaffected earnings account.

@Tecnativa
TT29658

ping @pedrobaeza @CarlosRoca13

CarlosRoca13 reviewed (commented):
Code review. I think that you can add tests to prove what you have corrected

joao-p-marques committed:
[FIX] account_financial_report: Don't include unaffected account if accounts are filtered

When generated a General Ledger or a Trial Balance, if some accounts are already filtered in the wizard don't show the information related to the unaffected earnings account

TT29658

joao-p-marques commented:
Tests added @CarlosRoca13 :+1:

pedrobaeza commented:
/ocabot merge patch

Please fw-port it to v14
